### PR TITLE
feat: Pending members selection

### LINKED ISSUE #9745: [CAL-2037] ability to add pending members to event-types
as an admin I want to

* invite people
* immediately add them to round robin, collective or managed event types without waiting them to accept

right now:

will not show up in "Assignment"

Only Daniel who is the only person who is not pending.

Pending people should not be actively participate in round robin or collective events but they should be able to be invited.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2037](https://linear.app/calcom/issue/CAL-2037/ability-to-add-pending-members-to-event-types)</sub>

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

Supports assigning pending members to organizations' sub-team events. Based on #9780, thanks for your work @nikospapcom.

Fixes #9745 

## Type of change

  - [X] New feature (non-breaking change which adds functionality)

## How should this be tested?

1. Create an organization
2. Create a few teams
3. Invite people to those teams
4. Create an event type on the teams you have pending invited members
5. See that you can select pending members (flagged as "pending")

## Mandatory Tasks

  - [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: yarn.lock
Comment (zomars) [inline yarn.lock:1]: No package.json were changed in this PR. Let's revert this file. Will update yarn lock if needed later

Comment (leog) [reply yarn.lock:1]: done @zomars

**Review by Udit-takkar — DISMISSED** [review]
Screenshot 2023-07-17 at 2 20 53 PM

getting forbidden when trying to save the event type after assigning to pending members

**Comment by leog** [comment]
> getting forbidden when trying to save the event type after assigning to pending members

Thanks @Udit-takkar, tackled the regular team scenario. We only want to let pending members to be assigned for organizations, at least for now. Asking @emrysal to review as well.

**Review by Udit-takkar — APPROVED** [review]

**Pull request merged**